From 5b76d69af88154252c31cf054145c03f49b0bcfe Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 17 May 2022 10:27:59 +0300
Subject: [PATCH 17/30] s32cc: Move u-boot-s32.cfgout target under BUILD_TARGET

This is the recommended way of adding additional targets
to default list.

Issue: ALB-8823
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 Makefile                         | 4 ----
 arch/arm/mach-s32/s32-cc/Kconfig | 3 +++
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 49e5ea6956..c790118a60 100644
--- a/Makefile
+++ b/Makefile
@@ -926,10 +926,6 @@ ifeq ($(CONFIG_ARCH_ROCKCHIP)$(CONFIG_SPL),yy)
 ALL-y += u-boot-rockchip.bin
 endif
 
-ifeq ($(CONFIG_NXP_S32CC),y)
-ALL-y += u-boot-s32.cfgout
-endif
-
 LDFLAGS_u-boot += $(LDFLAGS_FINAL)
 
 # Avoid 'Not enough room for program headers' error on binutils 2.28 onwards.
diff --git a/arch/arm/mach-s32/s32-cc/Kconfig b/arch/arm/mach-s32/s32-cc/Kconfig
index faa0759dbe..3d1fe62752 100644
--- a/arch/arm/mach-s32/s32-cc/Kconfig
+++ b/arch/arm/mach-s32/s32-cc/Kconfig
@@ -6,6 +6,9 @@ if NXP_S32CC
 config DEFAULT_DEVICE_TREE
 	default "s32cc"
 
+config BUILD_TARGET
+	default "u-boot-s32.cfgout"
+
 config CUSTOM_SYS_TEXT_BASE
 	bool "Use a custom SYS_TEXT_BASE"
 	help
-- 
2.17.1

